swipy/src/main/java/com/dotloop/swipy/SwipableCellHandler.kt

```diff
@@ -1,56 +1,49 @@
 package com.dotloop.swipy
 
 import com.dotloop.swipy.SwipableCellHandler.Mode.Single
-import java.util.*
 
-class SwipableCellHandler {
-
-    private val openPositions: MutableSet<Int> = HashSet()
-    private var mode = Single
+class SwipableCellHandler() {
 
+    private val openHistory: MutableMap<Int, SwipeLayout> = HashMap()
+    var mode = Single
 
     fun bindSwipableCell(swipeLayout: SwipeLayout, position: Int) {
         swipeLayout.tag = position
         swipeLayout.onSwipeListeners = (object : OnSwipeListener {
             override fun onClose(layout: SwipeLayout) {
-                openPositions.remove(layout.tag as Int)
+                openHistory.remove(layout.tag as Int)
             }
 
             override fun onStartOpen(layout: SwipeLayout) {
-                val position = layout.tag as Int
-                /**if (mode == Attributes.Mode.Single &&
-                 * !openPositions.isEmpty() &&
-                 * !openPositions.contains(position))
-                 * adapter.notifyItemChanged(position); */
+                val pos = layout.tag as Int
+                if (mode == Single && !openHistory.isEmpty() && !openHistory.containsKey(pos)) {
+                    openHistory.keys.forEach { openHistory[it]?.close(smooth = true, notify = true) }
+                }
             }
 
             override fun onOpen(layout: SwipeLayout) {
-                openPositions.add(layout.tag as Int)
+                openHistory[layout.tag as Int] = layout
             }
         })
 
-        swipeLayout.simpleOnLayoutChangeListenerListener = (object : SimpleOnLayoutChangeListener {
+        swipeLayout.simpleOnLayoutChangeListener = (object : SimpleOnLayoutChangeListener {
             override fun onLayoutChange(layout: SwipeLayout) =
-                if (isOpen(layout.tag as Int)) layout.open(false, false)
-                else layout.close(false, false)
+                if (isOpen(layout.tag as Int)) layout.open(smooth = false, notify = false)
+                else layout.close(smooth = false, notify = false)
         })
 
     }
 
     @JvmOverloads
     fun reset(position: Int? = null) {
```

Comment:
Is this method being used? If not, can you remove it?

Reply:
You are right, it's not used. You don't think it could be useful to have a method to programmatically closed on the "open" cell? Meaning cell being swiped.

Reply:
I think it is fine to leave it.